### PR TITLE
Progressive Web App Support

### PR DESCRIPTION
**Pull request by mankinskin**
#14

### CODE REVIEW

**Comment by emilk** [comment]
Interesting! This is the first time I hear about Progressive Web Apps. What, concretely, is made possible with this PR?

**Comment by mankinskin** [comment]
This PR allows the egui_template to be installed through a browser on the client, to make it accessible offline. For example on a smartphone the browser will ask the user to "Add site to homescreen". This will store the frontend locally like an app and it will be usable offline. When the user opens the "app", it will launch the website in a headless browser. The website can then use local offline state or make online requests. It works basically the same on desktops.

**Review by emilk — APPROVED** [review]

**Comment by emilk** [comment]
Very cool, thank you!

File: docs/sw.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+var cacheName = 'egui-template-pwa';
+var filesToCache = [
+  './',
+  './index.html',
+  './egui_template_bg.wasm'
```
Comment (emilk) [inline docs/sw.js:5]: Shouldn't `egui_template.js` be in here too?

Comment (mankinskin) [reply docs/sw.js:5]: Makes sense, good call.

**Pull request merged**